dalle2_pytorch/dalle2_pytorch.py

```diff
@@ -423,54 +423,77 @@ def forward(
         *,
         text_encodings,
         text_embed,
-        mask = None,
-        cond_drop_prob = 0.2
+        mask=None,
+        cond_drop_prob=0.1
     ):
-        batch, text_enc_len, device = image_embed.shape[0], text_encodings.shape[-2], image_embed.device
+        """
+        Complete one forward pass through the diffusion prior network.
 
-        # in section 2.2, last paragraph
-        # "... consisting of encoded text, CLIP text embedding, diffusion timestep embedding, noised CLIP image embedding, final embedding for prediction"
+        If the dictionary passed from the DiffusionPrior wrapper has the 
+        key:value pair text_encodings:None, then we will assume
+        we are doing "embedding only" training.
 
-        text_embed, image_embed = rearrange_many((text_embed, image_embed), 'b d -> b 1 d')
+        In the case of embedding only training we will drop the text
+        encoding from the tokens entirely and train on the remaining
+        tokens mentioned in section 2.2 of the DALLE2 paper.
+        """
 
-        # whether text embedding is used for conditioning depends on whether text encodings are available for attention (for classifier free guidance, even though it seems from the paper it was not used in the prior ddpm, as the objective is different)
-        # but let's just do it right
-
-        if exists(mask):
-            not_all_masked_out = mask.any(dim = -1)
-            mask = torch.cat((mask, rearrange(not_all_masked_out, 'b -> b 1')), dim = 1)
-
-        if exists(mask):
-            mask = F.pad(mask, (0, 2), value = True) # extend mask for text embedding, noised image embedding, time step embedding, and learned query
+        # grab some contextual information
+        batch, device = image_embed.shape[0], image_embed.device
 
+        # gather tokens for casual transformer
+        text_embed, image_embed = rearrange_many(
+            (text_embed, image_embed), 'b d -> b 1 d')
         time_embed = self.time_embeddings(diffusion_timesteps)
         time_embed = rearrange(time_embed, 'b d -> b 1 d')
+        learned_queries = repeat(self.learned_query, 'd -> b 1 d', b=batch)
+
+        # check for embedding only training
+        if text_encodings is not None:
```

Comment:
I opted to use a big if-statement block here to separate out the two possible training methods. I'm definitely open to more concise suggestions!

Reply:
yup, that's fine, another way would be to set text encodings to an empty tensor with the dimension at which to be concatenated to be 0